compiler/rustc_codegen_ssa/src/mir/constant.rs

```diff
@@ -84,11 +94,11 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                             };
                             bx.scalar_to_backend(prim, scalar, bx.immediate_backend_type(layout))
                         } else {
-                            bug!("simd shuffle field {:?}", field)
+                            bug!("field is not a scalar {:?}", field)
                         }
                     })
                     .collect();
-                bx.const_struct(&values, false)
```

Comment:
Oh, I didn't even realize this extra layer of a hack... we're creating a const_struct for the simd_shuffle argument and then taking it apart again to turn it into a const_vector.

Can you add a comment [here](https://github.com/rust-lang/rust/blob/84ac80f1921afc243d71fd0caaa4f2838c294102/compiler/rustc_codegen_ssa/src/mir/block.rs#L925) saying something like
```
// FIXME: the simd_shuffle argument is actually an array, not a vector, so we need this special
// hack to make sure it is passed as an immediate. We should pass the shuffle indices
// as a vector instead to avoid this hack.
```